pkg/sql/catalog/tabledesc/structured.go

```diff
@@ -276,6 +276,12 @@ func ForEachExprStringInTableDesc(descI catalog.TableDescriptor, f func(expr *st
 	doCheck := func(c *descpb.TableDescriptor_CheckConstraint) error {
 		return f(&c.Expr)
 	}
+	doUniqueWithoutIndex := func(c *descpb.UniqueWithoutIndexConstraint) error {
+		if !c.IsPartial() {
+			return nil
+		}
+		return f(&c.Predicate)
+	}
```

Comment:
It appears that we completely forgot about unique-without-index partial expressions here. Did we forget about them elsewhere in this package? Do we validate them properly? What would it take to remove the "experimental" designation for this feature?